SPEC.md

```diff
@@ -302,29 +273,89 @@ IEEE 754-2008 decimal floating point numbers
 
 ```kdl
 node (u8)123
-node prop=(regex)".*"
+node prop=(regex).*
 (published)date "1970-01-01"
 (contributor)person name="Foo McBar"
 ```
 
 ### String
 
-Strings in KDL represent textual [Values](#value), or unusual identifiers. A
-String is either a [Quoted String](#quoted-string) or a
-[Raw String](#raw-string). Quoted Strings may include escaped characters, while
-Raw Strings always contain only the literal characters that are present.
+Strings in KDL represent textual UTF-8 [Values](#value). A String is either an
+[Identifier String](#identifier-string) (like `foo`), a [Quoted String](#quoted-string) (like `"foo"`) or
+a [Raw String](#raw-string) (like `#"foo"#`). Identifier Strings let you write short, "single-word" strings with a minimum of syntax; Quoted Strings let you write strings with whitespace (including newlines!) or escapes; Raw Strings let you write strings with whitespace *but without escapes*, allowing you to not worry about the string's content containing anything that might look like an escape.
```

Comment:
Hey, I like the new barewords! I think it would be useful to document whether the different types of strings MUST be treated identically or not. In other words, we should specify whether the following MUST behave identically:

```
foo arg
foo "arg"
```

If they MAY be treated differently, then this could be useful for CLI-like syntaxes, such as

```
node1 --option  // treated as an "option"
node1 "--option"  // treated as a "positional argument"
```

Reply:
The intention is definitely that they are identical, just different syntaxes for a "string". All the existing kdl1 locations that allow idents and strings (type annotations, node names, property names) don't allow you to distinguish between the two syntaxes, any more than they let you distinguish between the various quoted string syntaxes. Extending this to attribute values and property values shouldn't change things.

If you do have a node where you want to have both string positional args and boolean flags, the recommended way is to make the flags into boolean properties. `node1 --option=#true` and `node1 "--option"` are completely distinguishable. (If you're not mixing these two things, tho, then just having the presence of an ident string indicate a flag is perfectly fine, like `node1 --option`.)

Reply:
Fair enough. In that case, I think it would be worth adding this to the spec. Something like:

"Implementations MUST treat the different string forms identically."

Reply:
that's not technically correct. A document-oriented parser might represent those differently, because they're represented differently in text and they want to maintain formatting and allow control of formatting. I'm not sure the spec needs any additional language beyond specifying "these two syntaxes represent the same data type," which is already expressed.